tests/baselines/reference/templateLiteralTypes3.js

```diff
@@ -124,11 +124,51 @@ type Schema = { a: { b: { c: number } } };
 declare function chain<F extends keyof Schema>(field: F | `${F}.${F}`): void;
 
 chain("a");
+
+// Repro from #46125
+
+function ff1<T extends string>(x: `foo-${string}`, y: `${string}-bar`, z: `baz-${string}`) {
+    if (x === y) {
+        x;  // `foo-${string}`
+    }
+    if (x === z) {  // Error
+    }
+}
+
+function ff2(x: string, y: `foo-${string}` | 'bar') {
+    if (x === y) {
+        x;  // `foo-${string}` | 'bar'
+    }
+}
+
+function ff3(x: string, y: `foo-${string}`) {
+    if (x === 'foo-test') {
+        x;  // 'foo-test'
+    }
+    if (y === 'foo-test') {
+        y;  // 'foo-test'
+    }
+}
+
+// Repro from #46045
+
+export type Action =
```

Comment:
We probably want to remove the `export` from this and the other line to keep around the declaration emit in the test.

Reply:
Yeah, for sure.